src/shell/commands/detect_hotkey.cpp

```diff
@@ -0,0 +1,116 @@
+#include "shell/commands.h"
+#include "shell/argh.h"
+#include "server/pegasus_read_service.h"
+
+bool validate_cmd(const argh::parser &cmd, const std::set<std::string> &params)
+{
+    if (cmd.size() > 1) {
+        fmt::print(stderr, "too many params!\n");
+        return false;
+    }
+
+    for (const auto &param : cmd.params()) {
+        if (params.find(param.first) == params.end()) {
+            fmt::print(stderr, "unknown param {} = {}\n", param.first, param.second);
+            return false;
+        }
+    }
+
+    return true;
+}
```

Comment:
The code is repeated with `disk_rebalance.cpp`, you can move it to `command_helper.h`